c10/core/inference_mode.cpp

```diff
@@ -0,0 +1,36 @@
+
+#include <c10/core/inference_mode.h>
+
+#include <stdexcept>
+
+namespace c10 {
+
+/// thread_local is a feature that is not enabled by Caffe2 mobile
+/// build (e.g. iOS). Therefore, we only provide `at::InferenceMode`
+/// when we are not in mobile build or when FEATURE_TORCH_MOBILE
+/// is on.
+#if !defined(C10_MOBILE) || defined(FEATURE_TORCH_MOBILE)
+
+thread_local bool InferenceMode_enabled = false;
```

Comment:
You sure about this being its own, separate thread local? Aren't you doubling the number of necessary TLS accesses now?